File: naïve.swift
func nthPrime(_ n: Int) -> Int {
	var i = 0
	var j = 2
	while i < n {
		if isPrime(j) {
			i += 1
		}
		j += 1
	}
	return j - 1
}

func isPrime(_ x: Int) -> Bool {
	if x < 2 {
		return false
	}
	var i = x - 1
	while i >= 2 {
		if x % i == 0 {
			return false
		}
		i -= 1
	}
	return true
}

print(nthPrime(10000))

File: naïve.go
package main

import (
	"fmt"
	"os"
	"runtime/pprof"
)

func nthPrime(n int32) int32 {
	var j, i int32
	j = 2
	for i = 0; i < n; {
		if isPrime(j) {
			i += 1
		}
		j += 1
	}
	return j - 1
}

func isPrime(n int32) bool {
	var i int32
	for i = 2; i < n; i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func main() {
	f, _ := os.Create("cpu.prof")
	pprof.StartCPUProfile(f)
	fmt.Printf("%v", nthPrime(10000))
	pprof.StopCPUProfile()
}
